### PR TITLE
Feature/circleci

### PR DESCRIPTION
**Pull request by JorgeMichelena**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+version: 2
+jobs:
+  build:
+    working_directory: ~/target/
+    docker:
+      - image: circleci/python:3.7.6-stretch-browsers
+        environment:
+          CIRCLECI: true
+          SECRET_KEY: '!f7k)2j8c!@o!ljx7deoaylhvv)1b$n(np-9l-x7_!4h#*ux*8'
```
Comment (MathiasLantean) [inline .circleci/config.yml:9]: You must avoid committing keys, this one, in particular, is used for encrypting user passwords so it's quite sensible. CircleCi allows you to define [environment variables ](https://circleci.com/docs/2.0/env-vars/).

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      - run: sudo pip3 install -r requirements.txt
+      - run: sudo apt-get install redis-server
+      - run: sudo redis-server --daemonize yes
+      - save_cache:
+          key: deps1-{{ .Branch }}-{{ checksum "requirements.txt" }}
+          paths:
+            - "/"
+      - run: flake8
+      - run: coverage run --source=. manage.py test
+      - run: coverage report
```
Comment (MathiasLantean) [inline .circleci/config.yml:40]: You should add the flag `--fail-under=90` or the code coverage percentage that you want to check. This flag is quite annoying but makes us doing tests 😂

File: chat/tests/test_chat.py [hunk 3]
```diff
@@ -34,13 +34,13 @@ def test_chat_message_posted_then_seen_in_different_windows_same_user_logged(sel
 
         self._switch_to_window(0)
         self._post_message('hello')
-        WebDriverWait(self.driver, 2).until(
+        WebDriverWait(self.driver, 4).until(
             lambda _:
             'hello' in self._chat_log_value(),
```
Comment (brunomichetti) [inline chat/tests/test_chat.py:39]: Can you change the lambda function for a condition? In this case I think that a function is not needed, but I don't know it the WebDriverWait allows that

File: targets/factory.py [hunk 4]
```diff
@@ -26,7 +26,7 @@ class Meta:
 class TargetFactory(Factory):
     factory.Faker.add_provider(DjangoGeoPointProvider)
 
-    title = factory.Faker('word')
+    title = factory.Sequence(lambda n: 'Target %d' % n)
```
Comment (brunomichetti) [inline targets/factory.py:29]: Why did you change this? I think that `Faker('word')` is more realistic

File: chat/tests/test_chat.py [hunk 5]
```diff
@@ -34,13 +34,13 @@ def test_chat_message_posted_then_seen_in_different_windows_same_user_logged(sel
...
             'Message was not received by window 1 from window 1'
                                             )
         self._switch_to_window(1)
-        WebDriverWait(self.driver, 2).until(
+        WebDriverWait(self.driver, 4).until(
```
Comment (brunomichetti) [inline chat/tests/test_chat.py:43]: Why did you change these numbers?

File: targets/factory.py (hunk 4 above)
Comment (kaozdl) [reply targets/factory.py:29]: i think that if titles need to be unique it makes more sense to have consistent tests than realistic ones, if that's not the thing tested with this.

File: pytest.ini [hunk 6]
```diff
@@ -0,0 +1,3 @@
+[pytest]
+DJANGO_SETTINGS_MODULE = target.settings
+python_files = test_chat.py test_unread_messages.py
```
Comment (MathiasLantean) [inline pytest.ini:3]: No line at the end of the file.
